cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/AbstractAssemblyOperator.java

```diff
@@ -98,12 +113,58 @@ protected static String name(HasMetadata resource) {
         return null;
     }
 
+    private final void reconcileCertificate(String namespace, Handler<AsyncResult<Void>> handler) {
+
+        vertx.createSharedWorkerExecutor("kubernetes-ops-pool").executeBlocking(
+                future -> {
+
+                    if (secretOperations.get(namespace, "internal-ca") == null) {
+                        log.info("Generating internal CA certificate ...");
+                        File internalCAkeyFile = null;
+                        File internalCAcertFile = null;
+                        try {
+                            CertManager certManager = new OpenSslCertManager();
+                            internalCAkeyFile = File.createTempFile("tls", "internal-ca-key");
+                            internalCAcertFile = File.createTempFile("tls", "internal-ca-cert");
+                            certManager.generateSelfSignedCert(internalCAkeyFile, internalCAcertFile, DEFAULT_CERTS_EXPIRATION_DAYS);
+
+                            SecretCertProvider secretCertProvider = new SecretCertProvider();
+                            Secret secret = secretCertProvider.createSecret(namespace, "internal-ca",
+                                    "internal-ca.key", "internal-ca.crt",
+                                    internalCAkeyFile, internalCAcertFile, Collections.emptyMap());
+
+                            secretOperations.reconcile(namespace, "internal-ca", secret)
+                                    .compose(future::complete, future);
+
+                        } catch (IOException e) {
+                            future.fail(e);
+                        } finally {
+                            if (internalCAkeyFile != null)
+                                internalCAkeyFile.delete();
+                            if (internalCAcertFile != null)
+                                internalCAcertFile.delete();
+                        }
+                        log.info("... end generating certificate");
+                    } else {
+                        log.info("The internal CA certificate already exists");
```

Comment:
Assuming this runs on every reconciliation (since the certificates will exist most of the time) ... is the _INFO_ level really needed? I think we should put it to lower log level.

Reply:
Agreed. I left this level by mistake. I needed it for testing.